app/graphql/cache.ts

```diff
@@ -61,6 +63,33 @@ const getWallets = ({
   return account.wallets
 }
 
+const getDisplayCurrencyAndFractionDigits = (cache: InMemoryCache) => {
+  // FIXME as RealtimePriceQuery is a singleton,
+  // could we have a way to not fetch DisplayCurrencyQuery?
+  const resCurrency = cache.readQuery<DisplayCurrencyQuery>({
+    query: DisplayCurrencyDocument,
+  })
+  const displayCurrency = resCurrency?.me?.defaultAccount?.displayCurrency
+
+  if (!displayCurrency) {
+    return { displayCurrency: null, fractionDigits: null }
+  }
+
+  const res2 = cache.readQuery<CurrencyListQuery>({
+    query: CurrencyListDocument,
+    variables: { currency: displayCurrency },
+  })
+  const currencyList = res2?.currencyList ?? []
+  if (!res2?.currencyList) {
+    return { displayCurrency: null, fractionDigits: null }
+  }
+
+  const fractionDigits =
```

Comment:
Does it make sense to have a default to 2  here, in other places if we can't find the value we return null. Something is wrong with the currency list if a value can't be found.